### PR TITLE
Add bzip2 port

### PR DESCRIPTION
**Pull request by msabramo**
so I don't need to create my own GitHub repo just so that I can get a `.zip` URL.

```
embuilder:INFO: building and verifying bzip2
system_libs:WARNING: retrieving port: bzip2 from 
    https://downloads.sourceforge.net/project/bzip2/bzip2-1.0.6.tar.gz
system_libs:WARNING: unpacking port: bzip2
cache:INFO: generating port: libbz2.a... 
    (this will be cached in "/root/.emscripten_cache/asmjs/libbz2.a" for subsequent builds)
cache:INFO:  - ok
embuilder:INFO: ...success
```

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -872,7 +872,15 @@ def fetch_project(name, url, subdir, is_tarbz2=False):
           Ports.clear_project_build(name)
           return
 
-    fullpath = fullname + ('.tar.bz2' if is_tarbz2 else '.zip')
+    appended_extension = False
+    for _, extensions, _ in shutil.get_unpack_formats():
```
Comment (sbc100) [inline tools/system_libs.py]: I think this is python3 only... can we just hardcode a list of possible extensions?

File: tools/system_libs.py [hunk 2]
```diff
@@ -909,16 +920,7 @@ def check_tag():
...
-        z = tarfile.open(fullpath, 'r:bz2')
-      else:
-        z = zipfile.ZipFile(fullpath, 'r')
-      try:
-        cwd = os.getcwd()
-        os.chdir(fullname)
-        z.extractall()
-      finally:
-        os.chdir(cwd)
+      shutil.unpack_archive(fullpath, fullname)
```
Comment (sbc100) [inline tools/system_libs.py]: Again python3 only I think.

Perhaps add a TODO here that when we switch to python3 we should convert this.

File: tests/bzip2_test.c [hunk 3]
```diff
@@ -0,0 +1,141 @@
+
```
[the remark this answers is not in the document]
Comment (msabramo) [reply tests/bzip2_test.c]: Done.

File: tools/system_libs.py (hunk 1 above)
Comment (msabramo) [reply tools/system_libs.py]: Done.

File: tools/system_libs.py (hunk 2 above)
Comment (msabramo) [reply tools/system_libs.py]: Done. Thanks!

**Comment by msabramo** [comment]
I'm guessing the firefox test failure here is a blip.

File: tools/system_libs.py [hunk 4]
```diff
@@ -872,7 +872,12 @@ def fetch_project(name, url, subdir, is_tarbz2=False):
           Ports.clear_project_build(name)
           return
 
-    fullpath = fullname + ('.tar.bz2' if is_tarbz2 else '.zip')
+    if is_tarbz2:
+        fullpath = fullname + '.tar.bz2'
```
[the remark this answers is not in the document]
Comment (msabramo) [reply tools/system_libs.py]: Fixed, thanks!

**Comment by msabramo** [comment]
Screen Shot 2019-03-27 at 6 53 43 PM

🎉

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**